src/atopile/front_end.py

```diff
@@ -838,29 +724,53 @@ def _new_node(
             # Descend into building the superclass. We've got no information
             # on when the super-chain will be resolved, so we need to promise
             # that this current blockdef will be visited as part of the init
-            new_node, supers_ = self._new_node(base_class, [item] + promised_supers)
+            if super_ctx:
+                from_ctxs_ = from_ctxs + [super_ctx]
+            else:
+                from_ctxs_ = from_ctxs
+
+            result = self._new_node(
+                base_class,
+                promised_supers=[item] + promised_supers,
+                from_ctxs=from_ctxs_,
+            )
 
             # HACK: promised_supers is falsey a hack way to check
             # that we're only looking at the top-node w/ promised supers
             if not promised_supers and (block_type.COMPONENT() or block_type.MODULE()):
-                new_node.add(has_ato_cmp_attrs())
+                result[0].add(has_ato_cmp_attrs())
```

Comment:
Switched to a tuple here because completely splitting and rejoining the results was unweildy